lib-cloudsdk/src/main/java/com/facebook/openwifi/cloudsdk/UCentralUtils.java

```diff
@@ -387,11 +387,11 @@ public static Map<String, Map<String, List<Integer>>> getDeviceAvailableChannels
 	 * Returns the results map
 	 */
 	public static Map<String, String> getBssidsMap(
-		Map<String, State> latestState
+		Map<String, StateInfo> latestState
 	) {
 		Map<String, String> bssidMap = new HashMap<>();
-		for (Map.Entry<String, State> e : latestState.entrySet()) {
-			State state = e.getValue();
+		for (Map.Entry<String, StateInfo> e : latestState.entrySet()) {
+			StateInfo state = e.getValue();
```

Comment:
You don't need to change this line if you don't actually need the subclass - keep the declaration as `State`. Same comment in the other files (`ModelerUtils` etc.) -- you don't need to change as many lines as you have.

Actually, you can make the least amount of changes if you modify the function parameters to use generics, like this:
```
Map<String, StateInfo>
-->
Map<String, ? extends State>  // only when you don't need StateInfo specific fields
```

Reply:
In `ModelerUtils`, `StateInfo` specific fields are needed. I change it in other files. Thanks for pointing out!